packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java

```diff
@@ -372,10 +372,34 @@ public void onConnected(Bundle connectionHint) {
    * properly initialize this listener.
    */
   @Override
-  public void onConnectionFailed(@NonNull ConnectionResult result) {
-    // We can attempt to reconnect if, e.g. the activity is paused and resumed.
-    if (pendingOperation != null && pendingOperation.method.equals(METHOD_INIT)) {
-      finishWithError(ERROR_REASON_CONNECTION_FAILED, result.toString());
+  public void onConnectionFailed(@NonNull final ConnectionResult result) {
+    if (resolvingError) {
+      // Already attempting to resolve an error.
+      return;
+    } else if (result.hasResolution()) {
+      try {
+        resolvingError = true;
+        result.startResolutionForResult(activity, REQUEST_CODE_RESOLVE_ERROR);
+      } catch (SendIntentException e) {
+        // There was an error with the resolution intent. Try again.
+        googleApiClient.connect();
+      }
+    } else {
+      resolvingError = true;
+      GoogleApiAvailability.getInstance()
+          .showErrorDialogFragment(
+              activity,
+              result.getErrorCode(),
+              REQUEST_CODE_RESOLVE_ERROR,
+              new DialogInterface.OnCancelListener() {
+                public void onCancel(DialogInterface dialog) {
+                  if (pendingOperation != null && pendingOperation.method.equals(METHOD_INIT)) {
```

Comment:
If we bring up an ErrorDialog, that usually means the user has to take action to fix it. If he/she cancels out of fixing the error, there is no point in attempting a re-connect. We'll just end up in the same error state.